draft-ietf-tls-esni.md

```diff
@@ -792,50 +811,49 @@ certificate validation.
 
 {{?I-D.ietf-tls-sni-encryption}} lists several requirements for SNI
 encryption. In this section, we re-iterate these requirements and assess
-the ESNI design against them.
+the ECHO design against them.
 
 ### Mitigate against replay attacks
 
-Since the SNI encryption key is derived from a (EC)DH operation
-between the client's ephemeral and server's semi-static ESNI key, the ESNI
-encryption is bound to the Client Hello. It is not possible for
-an attacker to "cut and paste" the ESNI value in a different Client
-Hello, with a different ephemeral key share, as the terminating server
-will fail to decrypt and verify the ESNI value.
+Since servers process either ClientHelloInner or ClientHelloOuter, and ClientHelloInner
+contains an HPKE-derived nonce, it is not possible for an attacker to "cut and paste"
+the ECHO value in a different Client Hello and learn information from ClientHelloInner.
+This is because attacker lacks access to the HPKE-derived nonce used to derive the
+handshake secrets.
```

Comment:
Is the server required to check that the nonces match?

Reply:
Doh. Good idea. It definitely should!